### ISSUE TITLE
Implement XDG base directories

### ISSUE THREAD

**Issue by mildred** [ipfs/kubo#120]
See: http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html

Basically:

configuration should be searched in:
- XDG_CONFIG_HOME/go-ipgs/config
- XDG_CONFIG_DIRS/go-ipgs/config

Some rules :
- XDG_CONFIG_HOME defaults to HOME/.config when not set
- XDG_CONFIG_DIRS defaults to  /etc/xdg when not set

The same logic applies to persistent data files (the one we want to keep) and cached data files (the one we may delete at any moment)

**Comment by Winterhuman** [comment]
@Stebalien Any chance this could be re-opened? Kubo is one of the few applications which doesn't store config files in `~/.config`

**Comment by Jorropo** [comment]
@Winterhuman actually any reason you can't just use `IPFS_PATH` ?

**Comment by Winterhuman** [comment]
@Jorropo Using `IPFS_PATH` wouldn't be the same, it's about making `~/.config/ipfs` be detected by default alongside `~/.ipfs` (basically the equivalent to `IPFS_PATH=.ipfs,.config/ipfs` as the default mode)

**Comment by Stebalien** [comment]
I'd _personally_ like this to happen, but I'd start out by defaulting to `~/.ipfs` for now so as not to break a bunch of tools (by default).

However, I won't be working on it. I'll open it up so it can be triaged, but it may be closed again as "not a priority"

**Comment by Jorropo** [comment]
This probably will be classed P4, Help wanted.

**Comment by alichtman** [comment]
Having `kubo` respect the XDG spec would be great. I try to keep my home directory clean, and the `$HOME/.ipfs` directory that's currently created by default could be placed elsewhere.

```
$ ls ~/.ipfs      
 blocks
 datastore
 keystore
 config
 datastore_spec
 version
```

`config`, `datastore_spec` and `version` all seem config-related, and could go in `$XDG_CONFIG_HOME/ipfs/`.

`blocks`, `datastore` and `keystore` could go in `$XDG_DATA_HOME/ipfs/`.

Keeping `~/.ifps` as the first path searched to maintain backwards compatibility makes sense.

For new installations (where `~/.ipfs` does not exist and `$IPFS_PATH` is unset), we could set these new defaults. And the migration path for users would look something like:

```bash
$ mkdir -p $XDG_CONFIG_HOME/ipfs
$ mkdir -p $XDG_DATA_HOME/ipfs
$ mv $HOME/.ipfs/config $HOME/.ipfs/datastore_spec $HOME/.ipfs/version $XDG_CONFIG_HOME/ipfs
$ mv $HOME/.ipfs/blocks $HOME/.ipfs/datastore $HOME/.ipfs/keystore $XDG_DATA_HOME/ipfs
```

**Comment by terashy88** [comment]
Hello everyone, I made a script that helps me set up IPFS.

By following the [XDG Base Directory](https://wiki.archlinux.org/title/XDG_Base_Directory)

I'm getting an error while using 
`IPFS_PATH="$XDG_DATA_HOME"/ipfs`

On `~/.ipfs` everything works fine.

On `"$XDG_DATA_HOME"/ipfs` `ipfs daemon` works as well.

By using systemctl status on the default Path `~/.ipfs` I get: 
```
❯ status "ipfs@"$USER".service"
● ipfs@rico.service - InterPlanetary File System (IPFS) daemon
     Loaded: loaded (/usr/lib/systemd/system/ipfs@.service; disabled; preset: disabled)
     Active: active (running) since Wed 2026-02-04 15:21:12 CET; 55s ago
 Invocation: 976f61f25b7c4eb98018e8d3490a6e4e
   Main PID: 122854 (ipfs)
      Tasks: 55 (limit: 37858)
     Memory: 260.6M (peak: 355.9M)
        CPU: 17.405s
     CGroup: /system.slice/system-ipfs.slice/ipfs@rico.service
             └─122854 /usr/bin/ipfs daemon

Feb 04 15:21:14 v15 ipfs[122854]: No data sent yet. To opt-out before collection starts:
Feb 04 15:21:14 v15 ipfs[122854]: • Set environment: IPFS_TELEMETRY=off
Feb 04 15:21:14 v15 ipfs[122854]: • Or run: ipfs config Plugins.Plugins.telemetry.Config.Mode off
Feb 04 15:21:14 v15 ipfs[122854]: • Then restart daemon
Feb 04 15:21:14 v15 ipfs[122854]: This message is shown only once.
Feb 04 15:21:14 v15 ipfs[122854]: Learn more: https://github.com/ipfs/kubo/blob/master/docs/telemetry.md
Feb 04 15:21:14 v15 ipfs[122854]: RPC API server listening on /ip4/127.0.0.1/tcp/5001
Feb 04 15:21:14 v15 ipfs[122854]: WebUI: http://127.0.0.1:5001/webui
Feb 04 15:21:14 v15 ipfs[122854]: Gateway server listening on /ip4/127.0.0.1/tcp/8080
Feb 04 15:21:14 v15 ipfs[122854]: Daemon is ready
```
And on Path `"$XDG_DATA_HOME"/ipfs` I would get:
```
❯ status "ipfs@"$USER".service"
× ipfs@rico.service - InterPlanetary File System (IPFS) daemon
     Loaded: loaded (/usr/lib/systemd/system/ipfs@.service; disabled; preset: disabled)
     Active: failed (Result: exit-code) since Wed 2026-02-04 15:01:31 CET; 17min ago
   Duration: 115ms
 Invocation: 0d384cebcbc8445a9e4427635eadd4ba
    Process: 118271 ExecStart=/usr/bin/ipfs daemon (code=exited, status=1/FAILURE)
   Main PID: 118271 (code=exited, status=1/FAILURE)
   Mem peak: 21.7M
        CPU: 86ms

Feb 04 15:01:31 v15 systemd[1]: ipfs@rico.service: Scheduled restart job, restart counter is at 5.
Feb 04 15:01:31 v15 systemd[1]: ipfs@rico.service: Start request repeated too quickly.
Feb 04 15:01:31 v15 systemd[1]: ipfs@rico.service: Failed with result 'exit-code'.
Feb 04 15:01:31 v15 systemd[1]: Failed to start InterPlanetary File System (IPFS) daemon.                                                                               
```

The script I made is:
```
ipfs_UP() {
    # https://wiki.archlinux.org/title/InterPlanetary_File_System
    # Needed for Browser IPFS Extension

    pkg_=(
        kubo
    )

    service="ipfs@$USER.service"

    # path_=~/.ipfs
    path_="$XDG_DATA_HOME"/ipfs

    if ! pacman -Qq "$pkg_" >/dev/null 2>&1; then
        if [[ $(pacman -Su --print-format "%v" | wc -l) -gt 0 ]]; then
            sudo pacman -Sy
        fi
        for pkg in "${pkg_[@]}"; do

            if ! pacman -Qq "$pkg" >/dev/null 2>&1; then
                sudo pacman -Su "$pkg"
            fi
        done
    fi

    if ! [[ -d "$path_" ]] >/dev/null 2>&1; then

        # mkdir -p "$path_"

        export IPFS_PATH="$path_"

        ipfs init
        # IPFS_PATH="$path_" ipfs init
        #    ipfs daemon &
        export IPFS_TELEMETRY=off
    else
        #     #    ipfs daemon &
        export IPFS_TELEMETRY=off
        ls -lah "$path_"
    fi

    if systemctl status "$service" | grep "inactive" >/dev/null 2>&1; then
        sudo systemctl enable "$service" --now
    elif systemctl status "$service" | grep "failed" >/dev/null 2>&1; then
        sudo systemctl start "$service" --now
    else
        systemctl status "$service"
    fi
}
```
